moby/sys
# ci/gha: add go 1.16; run as root; fix some tests; add fedora

A few tiny improvements to GHA CI:
 - add go 1.16 to testing matix :fire: ;
 - explicitly specify `ubuntu-20.04` (rather than `ubuntu-latest`);
 - move 'make lint' and 'make cross' to before 'make test';
 - update actions/setup-go to v2;
 - add :womans_hat: fedora  runner (to test functionality that needs `openat2(2)`);
 - remove remaining travis-ci bits;
 - fix mount tests for kernel 5.9+;
 - run tests as root :fire:. 

The last item revealed that tests that require `openat2` do not check for it :see_no_evil: :, so add the check.
